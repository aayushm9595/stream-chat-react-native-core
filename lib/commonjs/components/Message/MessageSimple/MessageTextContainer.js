var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageTextContainer = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));

var _react = _interopRequireWildcard(require("react"));

var _capitalize = require("./utils/capitalize");

var _renderText = require("./utils/renderText");

var _styledComponents = require("../../../styles/styledComponents");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/Message/MessageSimple/MessageTextContainer.tsx";

function _templateObject() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  align-self: ", ";\n  background-color: ", ";\n  border-bottom-left-radius: ", "px;\n  border-bottom-right-radius: ", "px;\n  border-color: ", ";\n  border-top-left-radius: ", "px;\n  border-top-right-radius: ", "px;\n  border-width: ", "px;\n  margin-top: 2px;\n  padding-horizontal: 8px;\n  padding-vertical: 5px;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var TextContainer = _styledComponents.styled.View(_templateObject(), function (_ref) {
  var alignment = _ref.alignment;
  return alignment === 'left' ? 'flex-start' : 'flex-end';
}, function (_ref2) {
  var alignment = _ref2.alignment,
      status = _ref2.status,
      theme = _ref2.theme,
      type = _ref2.type;
  return alignment === 'left' || type === 'error' || status === 'failed' ? theme.colors.transparent : theme.colors.light;
}, function (_ref3) {
  var groupStyle = _ref3.groupStyle,
      theme = _ref3.theme;
  return groupStyle.indexOf('left') !== -1 ? theme.message.content.textContainer.borderRadiusS : theme.message.content.textContainer.borderRadiusL;
}, function (_ref4) {
  var groupStyle = _ref4.groupStyle,
      theme = _ref4.theme;
  return groupStyle.indexOf('right') !== -1 ? theme.message.content.textContainer.borderRadiusS : theme.message.content.textContainer.borderRadiusL;
}, function (_ref5) {
  var alignment = _ref5.alignment,
      theme = _ref5.theme;
  return alignment === 'left' ? theme.message.content.textContainer.leftBorderColor : theme.message.content.textContainer.rightBorderColor;
}, function (_ref6) {
  var groupStyle = _ref6.groupStyle,
      theme = _ref6.theme;
  return groupStyle === 'leftBottom' || groupStyle === 'leftMiddle' ? theme.message.content.textContainer.borderRadiusS : theme.message.content.textContainer.borderRadiusL;
}, function (_ref7) {
  var groupStyle = _ref7.groupStyle,
      theme = _ref7.theme;
  return groupStyle === 'rightBottom' || groupStyle === 'rightMiddle' ? theme.message.content.textContainer.borderRadiusS : theme.message.content.textContainer.borderRadiusL;
}, function (_ref8) {
  var alignment = _ref8.alignment,
      theme = _ref8.theme;
  return alignment === 'left' ? theme.message.content.textContainer.leftBorderWidth : theme.message.content.textContainer.rightBorderWidth;
}, function (_ref9) {
  var theme = _ref9.theme;
  return theme.message.content.textContainer.css;
});

var MessageTextContainer = function MessageTextContainer(props) {
  var alignment = props.alignment,
      _props$groupStyles = props.groupStyles,
      groupStyles = _props$groupStyles === void 0 ? ['bottom'] : _props$groupStyles,
      _props$markdownRules = props.markdownRules,
      markdownRules = _props$markdownRules === void 0 ? {} : _props$markdownRules,
      message = props.message,
      MessageText = props.MessageText;
  var theme = (0, _react.useContext)(_styledComponents.ThemeContext);
  if (!message.text) return null;
  var groupStyle = alignment + (0, _capitalize.capitalize)(Array.isArray(message.attachments) && message.attachments.length > 0 ? 'bottom' : groupStyles[0]);
  var markdownStyles = theme ? theme.message.content.markdown : {};
  return _react["default"].createElement(TextContainer, {
    alignment: alignment,
    groupStyle: groupStyle,
    status: message.status,
    testID: "message-text-container",
    type: message.type,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 170,
      columnNumber: 5
    }
  }, MessageText ? _react["default"].createElement(MessageText, (0, _extends2["default"])({}, props, {
    renderText: _renderText.renderText,
    theme: theme,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 178,
      columnNumber: 9
    }
  })) : (0, _renderText.renderText)({
    markdownRules: markdownRules,
    markdownStyles: markdownStyles,
    message: message
  }));
};

exports.MessageTextContainer = MessageTextContainer;
//# sourceMappingURL=MessageTextContainer.js.map