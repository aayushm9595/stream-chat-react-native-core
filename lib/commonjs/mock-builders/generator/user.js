var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateStaticUser = exports.generateUser = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _uuid = require("uuid");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var generateUser = function generateUser() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    banned: false,
    created_at: '2020-04-27T13:39:49.331742Z',
    id: (0, _uuid.v4)(),
    image: (0, _uuid.v4)(),
    name: (0, _uuid.v4)(),
    online: false,
    role: 'user',
    updated_at: '2020-04-27T13:39:49.332087Z'
  }, options);
};

exports.generateUser = generateUser;

var generateStaticUser = function generateStaticUser(userNumber) {
  switch (userNumber) {
    case 0:
      return generateUser({
        id: 'DannyBoi1',
        image: 'https://images-na.ssl-images-amazon.com/images/I/51t29lLkg8L._AC_SL1000_.jpg',
        name: 'Dan'
      });

    case 1:
      return generateUser({
        id: 'GrantyBoi2',
        image: 'https://i.imgur.com/SLx06PP.png',
        name: 'Grant'
      });

    case 2:
      return generateUser({
        id: 'ViriBoi2',
        image: 'https://i.imgur.com/iNaC3K7.jpg',
        name: 'Vir'
      });

    default:
      return generateUser({
        id: 'NeiliBoi3',
        image: 'https://i.imgur.com/T68W8nR_d.webp?maxwidth=728&fidelity=grand',
        name: 'Neil'
      });
  }
};

exports.generateStaticUser = generateStaticUser;
//# sourceMappingURL=user.js.map