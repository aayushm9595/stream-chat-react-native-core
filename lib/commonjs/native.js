Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registerNativeHandlers = exports.pickDocument = exports.pickImage = exports.NetInfo = void 0;

var fail = function fail() {
  throw Error('Native handler was not registered, you should import stream-chat-expo or stream-chat-react-native');
};

var NetInfo = {
  addEventListener: fail,
  fetch: fail
};
exports.NetInfo = NetInfo;
var pickImage = fail;
exports.pickImage = pickImage;
var pickDocument = fail;
exports.pickDocument = pickDocument;

var registerNativeHandlers = function registerNativeHandlers(handlers) {
  if (handlers.NetInfo) {
    exports.NetInfo = NetInfo = handlers.NetInfo;
  }

  if (handlers.pickImage) {
    exports.pickImage = pickImage = handlers.pickImage;
  }

  if (handlers.pickDocument) {
    exports.pickDocument = pickDocument = handlers.pickDocument;
  }
};

exports.registerNativeHandlers = registerNativeHandlers;
//# sourceMappingURL=native.js.map