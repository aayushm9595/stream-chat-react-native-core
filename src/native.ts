import type { NetInfoSubscription } from '@react-native-community/netinfo';

const fail = () => {
  throw Error(
    'Native handler was not registered, you should import stream-chat-expo or stream-chat-react-native',
  );
};

type NetInfo = {
  addEventListener: (
    listener: (isConnected: boolean) => void,
  ) => NetInfoSubscription | never;
  fetch: (requestedInterface?: string | undefined) => Promise<boolean> | never;
};

export let NetInfo: NetInfo = {
  addEventListener: fail,
  fetch: fail,
};

type PickImage = ({
  compressImageQuality,
  maxNumberOfFiles,
}: {
  compressImageQuality?: number;
  maxNumberOfFiles?: number;
}) => Promise<{ cancelled: boolean; images?: { uri: string }[] }> | never;
export let pickImage: PickImage = fail;

type PickDocument = ({
  maxNumberOfFiles,
}: {
  maxNumberOfFiles?: number;
}) =>
  | Promise<{
      cancelled: boolean;
      docs?: {
        name: string;
        size?: number | string;
        type?: string;
        uri?: string;
      }[];
    }>
  | never;
export let pickDocument: PickDocument = fail;

type Handlers = {
  NetInfo?: NetInfo;
  pickDocument?: PickDocument;
  pickImage?: PickImage;
};
export const registerNativeHandlers = (handlers: Handlers) => {
  if (handlers.NetInfo) {
    NetInfo = handlers.NetInfo;
  }

  if (handlers.pickImage) {
    pickImage = handlers.pickImage;
  }

  if (handlers.pickDocument) {
    pickDocument = handlers.pickDocument;
  }
};
