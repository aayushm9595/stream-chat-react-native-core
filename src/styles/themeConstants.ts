import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
import type { CSSObject } from 'styled-components';

export const BASE_FONT_SIZE = 16;

export const Colors = {
  danger: '#EDD8DD',
  light: '#EBEBEB',
  primary: '#006cff',
  secondary: '#111',
  textDark: 'rgba(0,0,0,1)',
  textGrey: 'rgba(0,0,0,0.5)',
  textLight: 'white',
  transparent: 'transparent',
};

export type CSS = {
  css?: CSSObject;
};

export type MarkdownStyle = Partial<{
  blockQuoteBar: StyleProp<ViewStyle>;
  blockQuoteSection: StyleProp<ViewStyle>;
  blockQuoteSectionBar: StyleProp<ViewStyle>;
  blockQuoteText: StyleProp<ViewStyle>;
  codeBlock: StyleProp<TextStyle>;
  del: StyleProp<TextStyle>;
  em: StyleProp<TextStyle>;
  heading: StyleProp<TextStyle>;
  heading1: StyleProp<TextStyle>;
  heading2: StyleProp<TextStyle>;
  heading3: StyleProp<TextStyle>;
  heading4: StyleProp<TextStyle>;
  heading5: StyleProp<TextStyle>;
  heading6: StyleProp<TextStyle>;
  hr: StyleProp<ViewStyle>;
  image: StyleProp<ImageStyle>;
  inlineCode: StyleProp<TextStyle>;
  link: StyleProp<TextStyle>;
  listItem: StyleProp<ViewStyle>;
  listItemBulletType: string;
  listItemButton: StyleProp<TextStyle>;
  listItemNumber: StyleProp<TextStyle>;
  listItemText: StyleProp<TextStyle>;
  mailTo: StyleProp<TextStyle>;
  paragraph: StyleProp<ViewStyle>;
  strong: StyleProp<TextStyle>;
  table: StyleProp<ViewStyle>;
  tableHeader: StyleProp<ViewStyle>;
  tableHeaderCell: StyleProp<ViewStyle>;
  tableRow: StyleProp<ViewStyle>;
  tableRowCell: StyleProp<ViewStyle>;
  tableRowLast: StyleProp<ViewStyle>;
  text: StyleProp<TextStyle>;
  u: StyleProp<TextStyle>;
  video: StyleProp<ImageStyle>;
  view: StyleProp<ViewStyle>;
}>;

export type Theme = {
  avatar: {
    container: CSS;
    fallback: CSS;
    image: CSS;
    text: CSS;
  };
  channelListFooterLoadingIndicator: {
    container: CSS;
  };
  channelListHeaderErrorIndicator: {
    container: CSS;
    errorText: CSS;
  };
  channelListHeaderNetworkDownIndicator: {
    container: CSS;
    errorText: CSS;
  };
  channelPreview: {
    container: CSS;
    date: CSS;
    details: CSS;
    detailsTop: CSS;
    message: CSS & {
      color: string;
      fontWeight: string;
      unreadColor: string;
      unreadFontWeight: string;
    };
    title: CSS;
  };
  closeButton: {
    container: CSS;
  };
  colors: typeof Colors;
  iconBadge: {
    container: CSS;
    icon: CSS;
    iconInner: CSS;
    unreadCount: CSS;
  };
  iconSquare: {
    container: CSS;
  };
  loadingErrorIndicator: {
    container: CSS;
    errorText: CSS;
    retryText: CSS;
  };
  loadingIndicator: {
    container: CSS;
    loadingText: CSS;
  };
  message: {
    actions: {
      button: CSS & {
        defaultBackgroundColor: string;
        defaultBorderColor: string;
        primaryBackgroundColor: string;
        primaryBorderColor: string;
      };
      buttonText: CSS & {
        defaultColor: string;
        primaryColor: string;
      };
      container: CSS;
    };
    actionSheet: {
      buttonContainer: CSS;
      buttonText: CSS;
      cancelButtonContainer: CSS;
      cancelButtonText: CSS;
      titleContainer: CSS;
      titleText: CSS;
    };
    avatarWrapper: {
      container: CSS;
      spacer: CSS;
    };
    card: {
      container: CSS;
      cover: CSS;
      footer: CSS & {
        description: CSS;
        link: CSS;
        logo: CSS;
        title: CSS;
      };
    };
    container: CSS;
    content: {
      container: CSS & {
        borderRadiusL: number;
        borderRadiusS: number;
      };
      containerInner: CSS;
      deletedContainer: CSS;
      deletedText: CSS;
      errorContainer: CSS & {
        backgroundColor: string;
      };
      // Available options for styling text: https://github.com/CharlesMangwa/react-native-simple-markdown/tree/next#styles-1
      markdown: MarkdownStyle;
      metaContainer: CSS;
      metaText: CSS;
      textContainer: CSS & {
        borderRadiusL: number;
        borderRadiusS: number;
        leftBorderColor: string;
        leftBorderWidth: number;
        rightBorderColor: string;
        rightBorderWidth: number;
      };
    };
    file: {
      container: CSS;
      details: CSS;
      icon: CSS;
      size: CSS;
      title: CSS;
    };
    gallery: {
      doubleSize: number;
      galleryContainer: CSS;
      halfSize: number;
      header: {
        button: CSS;
        container: CSS;
      };
      imageContainer: CSS;
      single: CSS;
      size: number;
      width: number;
    };
    reactionList: {
      container: CSS;
      reactionCount: CSS;
    };
    reactionPicker: {
      column: CSS;
      container: CSS;
      containerView: CSS;
      emoji: CSS;
      reactionCount: CSS;
      text: CSS;
    };
    replies: {
      container: CSS;
      image: CSS;
      messageRepliesText: CSS;
    };
    status: {
      checkMark: CSS;
      deliveredCircle: CSS;
      deliveredContainer: CSS;
      readByContainer: CSS;
      readByCount: CSS;
      sendingContainer: CSS;
      sendingImage: CSS;
      spacer: CSS;
    };
  };
  messageInput: {
    actionSheet: {
      buttonContainer: CSS;
      buttonText: CSS;
      titleContainer: CSS;
      titleText: CSS;
    };
    attachButton: CSS;
    attachButtonIcon: CSS;
    container: CSS & {
      conditionalPadding: number;
    };
    editingBoxContainer: CSS;
    editingBoxHeader: CSS;
    editingBoxHeaderTitle: CSS;
    fileUploadPreview: {
      attachmentContainerView: CSS;
      attachmentView: CSS;
      container: CSS;
      dismiss: CSS;
      dismissImage: CSS;
      filenameText: CSS;
      itemContainer: CSS;
    };
    imageUploadPreview: {
      container: CSS;
      dismiss: CSS;
      dismissImage: CSS;
      itemContainer: CSS;
      upload: CSS;
    };
    inputBox: CSS;
    inputBoxContainer: CSS;
    sendButton: CSS;
    sendButtonIcon: CSS;
    suggestions: {
      command: {
        args: CSS;
        container: CSS;
        description: CSS;
        title: CSS;
        top: CSS;
      };
      container: CSS & {
        itemHeight: number;
        maxHeight: number;
      };
      item: CSS;
      mention: {
        container: CSS;
        name: CSS;
      };
      separator: CSS;
      title: CSS;
      wrapper: CSS;
    };
    uploadProgressIndicator: {
      container: CSS;
      overlay: CSS;
    };
  };
  messageList: {
    dateSeparator: {
      container: CSS;
      date: CSS;
      dateText: CSS;
      line: CSS;
    };
    errorNotification: CSS;
    errorNotificationText: CSS;
    eventIndicator: {
      date: CSS;
      memberUpdateContainer: CSS;
      memberUpdateText: CSS;
      memberUpdateTextContainer: CSS;
    };
    listContainer: CSS;
    messageNotification: {
      container: CSS;
      text: CSS;
    };
    messageSystem: {
      container: CSS;
      dateText: CSS;
      line: CSS;
      text: CSS;
      textContainer: CSS;
    };
    typingIndicatorContainer: CSS;
  };
  spinner: CSS;
  thread: {
    newThread: CSS & {
      text: CSS;
    };
  };
  typingIndicator: {
    container: CSS;
    text: CSS & {
      color: string;
      fontSize: number;
    };
  };
};

export const defaultTheme: Theme = {
  avatar: {
    container: {},
    fallback: {},
    image: {},
    text: {},
  },
  channelListFooterLoadingIndicator: {
    container: {},
  },
  channelListHeaderErrorIndicator: {
    container: {},
    errorText: {},
  },
  channelListHeaderNetworkDownIndicator: {
    container: {},
    errorText: {},
  },
  channelPreview: {
    container: {},
    date: {},
    details: {},
    detailsTop: {},
    message: {
      color: '#767676',
      fontWeight: 'normal',
      unreadColor: '#000',
      unreadFontWeight: 'bold',
    },
    title: {},
  },
  closeButton: {
    container: {},
  },
  colors: {
    ...Colors,
  },
  iconBadge: {
    container: {},
    icon: {},
    iconInner: {},
    unreadCount: {},
  },
  iconSquare: {
    container: {},
  },
  loadingErrorIndicator: {
    container: {},
    errorText: {},
    retryText: {},
  },
  loadingIndicator: {
    container: {},
    loadingText: {},
  },
  message: {
    actions: {
      button: {
        defaultBackgroundColor: 'white',
        defaultBorderColor: 'transparent',
        primaryBackgroundColor: Colors.primary,
        primaryBorderColor: Colors.light,
      },
      buttonText: {
        defaultColor: 'black',
        primaryColor: 'white',
      },
      container: {},
    },
    actionSheet: {
      buttonContainer: {},
      buttonText: {},
      cancelButtonContainer: {},
      cancelButtonText: {},
      titleContainer: {},
      titleText: {},
    },
    avatarWrapper: {
      container: {},
      spacer: {},
    },
    card: {
      container: {},
      cover: {},
      footer: {
        description: {},
        link: {},
        logo: {},
        title: {},
      },
    },
    container: {},
    content: {
      container: {
        borderRadiusL: 16,
        borderRadiusS: 2,
      },
      containerInner: {},
      deletedContainer: {},
      deletedText: {},
      errorContainer: {
        backgroundColor: Colors.danger,
      },
      markdown: {},
      metaContainer: {},
      metaText: {},
      textContainer: {
        borderRadiusL: 16,
        borderRadiusS: 2,
        leftBorderColor: 'rgba(0, 0, 0, 0.08)',
        leftBorderWidth: 0.5,
        rightBorderColor: 'transparent',
        rightBorderWidth: 0,
      },
    },
    file: {
      container: {},
      details: {},
      icon: {},
      size: {},
      title: {},
    },
    gallery: {
      doubleSize: 240,
      galleryContainer: {},
      halfSize: 80,
      header: {
        button: {},
        container: {},
      },
      imageContainer: {},
      single: {},
      size: 120,
      width: 240,
    },
    reactionList: {
      container: {},
      reactionCount: {},
    },
    reactionPicker: {
      column: {},
      container: {},
      containerView: {},
      emoji: {},
      reactionCount: {},
      text: {},
    },
    replies: {
      container: {},
      image: {},
      messageRepliesText: {},
    },
    status: {
      checkMark: {},
      deliveredCircle: {},
      deliveredContainer: {},
      readByContainer: {},
      readByCount: {},
      sendingContainer: {},
      sendingImage: {},
      spacer: {},
    },
  },
  messageInput: {
    actionSheet: {
      buttonContainer: {},
      buttonText: {},
      titleContainer: {},
      titleText: {},
    },
    attachButton: {},
    attachButtonIcon: {},
    container: {
      conditionalPadding: 20,
    },
    editingBoxContainer: {},
    editingBoxHeader: {},
    editingBoxHeaderTitle: {},
    fileUploadPreview: {
      attachmentContainerView: {},
      attachmentView: {},
      container: {},
      dismiss: {},
      dismissImage: {},
      filenameText: {},
      itemContainer: {},
    },
    imageUploadPreview: {
      container: {},
      dismiss: {},
      dismissImage: {},
      itemContainer: {},
      upload: {},
    },
    inputBox: {},
    inputBoxContainer: {},
    sendButton: {},
    sendButtonIcon: {},
    suggestions: {
      command: {
        args: {},
        container: {},
        description: {},
        title: {},
        top: {},
      },
      container: {
        itemHeight: 50,
        maxHeight: 250,
      },
      item: {},
      mention: {
        container: {},
        name: {},
      },
      separator: {},
      title: {},
      wrapper: {},
    },
    uploadProgressIndicator: {
      container: {},
      overlay: {},
    },
  },
  messageList: {
    dateSeparator: {
      container: {},
      date: {},
      dateText: {},
      line: {},
    },
    errorNotification: {},
    errorNotificationText: {},
    eventIndicator: {
      date: {},
      memberUpdateContainer: {},
      memberUpdateText: {},
      memberUpdateTextContainer: {},
    },
    listContainer: {},
    messageNotification: {
      container: {},
      text: {},
    },
    messageSystem: {
      container: {},
      dateText: {},
      line: {},
      text: {},
      textContainer: {},
    },
    typingIndicatorContainer: {},
  },
  spinner: {},
  thread: {
    newThread: {
      text: {},
    },
  },
  typingIndicator: {
    container: {},
    text: {
      color: Colors.textGrey,
      fontSize: 14,
    },
  },
};
