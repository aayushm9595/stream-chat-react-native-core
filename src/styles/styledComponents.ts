import * as styledComponents from 'styled-components/native';

import type { Consumer, Context } from 'react';

import type { Theme } from './themeConstants';

declare module 'styled-components/native' {
  export interface DefaultTheme extends Theme {}
}

const appStyledComponents = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<
  Theme
>;

const {
  css,
  default: styled,
  ThemeProvider,
  useTheme,
  withTheme,
} = appStyledComponents;

const ThemeConsumer = appStyledComponents.ThemeConsumer as Consumer<Theme>;
const ThemeContext = appStyledComponents.ThemeContext as Context<Theme>;

export {
  css,
  styled,
  ThemeConsumer,
  ThemeContext,
  ThemeProvider as StyledComponentsThemeProvider,
  useTheme,
  withTheme,
};
