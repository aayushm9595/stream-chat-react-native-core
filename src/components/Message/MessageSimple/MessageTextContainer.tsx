import React, { useContext } from 'react';

import { capitalize } from './utils/capitalize';
import { renderText } from './utils/renderText';

import { styled, ThemeContext } from '../../../styles/styledComponents';

import type { MessageSimpleProps } from './MessageSimple';
import type { RenderTextParams } from './utils/renderText';

import type {
  Alignment,
  GroupType,
} from '../../../contexts/messagesContext/MessagesContext';
import type { Message as MessageType } from '../../../components/MessageList/utils/insertDates';
import type { Theme } from '../../../styles/themeConstants';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';

const TextContainer = styled.View<{
  alignment: Alignment;
  groupStyle: string;
  status?: string;
  type?: string;
}>`
  align-self: ${({ alignment }) =>
    alignment === 'left' ? 'flex-start' : 'flex-end'};
  background-color: ${({ alignment, status, theme, type }) =>
    alignment === 'left' || type === 'error' || status === 'failed'
      ? theme.colors.transparent
      : theme.colors.light};
  border-bottom-left-radius: ${({ groupStyle, theme }) =>
    groupStyle.indexOf('left') !== -1
      ? theme.message.content.textContainer.borderRadiusS
      : theme.message.content.textContainer.borderRadiusL}px;
  border-bottom-right-radius: ${({ groupStyle, theme }) =>
    groupStyle.indexOf('right') !== -1
      ? theme.message.content.textContainer.borderRadiusS
      : theme.message.content.textContainer.borderRadiusL}px;
  border-color: ${({ alignment, theme }) =>
    alignment === 'left'
      ? theme.message.content.textContainer.leftBorderColor
      : theme.message.content.textContainer.rightBorderColor};
  border-top-left-radius: ${({ groupStyle, theme }) =>
    groupStyle === 'leftBottom' || groupStyle === 'leftMiddle'
      ? theme.message.content.textContainer.borderRadiusS
      : theme.message.content.textContainer.borderRadiusL}px;
  border-top-right-radius: ${({ groupStyle, theme }) =>
    groupStyle === 'rightBottom' || groupStyle === 'rightMiddle'
      ? theme.message.content.textContainer.borderRadiusS
      : theme.message.content.textContainer.borderRadiusL}px;
  border-width: ${({ alignment, theme }) =>
    alignment === 'left'
      ? theme.message.content.textContainer.leftBorderWidth
      : theme.message.content.textContainer.rightBorderWidth}px;
  margin-top: 2px;
  padding-horizontal: 8px;
  padding-vertical: 5px;
  ${({ theme }) => theme.message.content.textContainer.css}
`;

export type MessageTextProps<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = MessageTextContainerProps<At, Ch, Co, Ev, Me, Re, Us> & {
  renderText: (
    params: RenderTextParams<At, Ch, Co, Ev, Me, Re, Us>,
  ) => JSX.Element | null;
  theme: Theme;
};

export type MessageTextContainerProps<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  /**
   * Position of the message, either 'right' or 'left'
   */
  alignment: Alignment;
  /**
   * Whether or not the message has failed
   */
  disabled: boolean;
  /**
   * Position of message in group - top, bottom, middle, single.
   *
   * Message group is a group of consecutive messages from same user. groupStyles can be used to style message as per their position in message group
   * e.g., user avatar (to which message belongs to) is only showed for last (bottom) message in group.
   */
  groupStyles: GroupType[];
  /** Handler to process a reaction */
  handleReaction: (reactionType: string) => Promise<void>;
  /**
   * Returns true if message belongs to current user, else false
   */
  isMyMessage: (message: MessageType<At, Ch, Co, Ev, Me, Re, Us>) => boolean;
  /**
   * Current [message object](https://getstream.io/chat/docs/#message_format)
   */
  message: MessageType<At, Ch, Co, Ev, Me, Re, Us>;
  /**
   * Handler to open and navigate into a message thread
   */
  openThread: () => void;
  /** Object specifying rules defined within simple-markdown https://github.com/Khan/simple-markdown#adding-a-simple-extension */
  markdownRules?: UnknownType;
  /**
   * Custom UI component to display a message in MessageList component
   * Default component (accepts the same props): [MessageSimple](https://getstream.github.io/stream-chat-react-native/#messagesimple)
   * */
  Message?: React.ComponentType<MessageSimpleProps<At, Ch, Co, Ev, Me, Re, Us>>;
  /**
   * Custom UI component for message text
   */
  MessageText?: React.ComponentType<
    MessageTextProps<At, Ch, Co, Ev, Me, Re, Us>
  >;
};

export const MessageTextContainer = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  props: MessageTextContainerProps<At, Ch, Co, Ev, Me, Re, Us>,
) => {
  const {
    alignment,
    groupStyles = ['bottom'],
    markdownRules = {},
    message,
    MessageText,
  } = props;
  const theme = useContext(ThemeContext);

  if (!message.text) return null;

  const groupStyle =
    alignment +
    capitalize(
      Array.isArray(message.attachments) && message.attachments.length > 0
        ? 'bottom'
        : groupStyles[0],
    );
  const markdownStyles = theme ? theme.message.content.markdown : {};

  return (
    <TextContainer
      alignment={alignment}
      groupStyle={groupStyle}
      status={message.status}
      testID='message-text-container'
      type={message.type}
    >
      {MessageText ? (
        <MessageText {...props} renderText={renderText} theme={theme} />
      ) : (
        renderText<At, Ch, Co, Ev, Me, Re, Us>({
          markdownRules,
          markdownStyles,
          message,
        })
      )}
    </TextContainer>
  );
};
