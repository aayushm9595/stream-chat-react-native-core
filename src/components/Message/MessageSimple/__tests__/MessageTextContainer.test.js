import React from 'react';
import { Text } from 'react-native';
import { cleanup, render, waitFor } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import {
  generateMessage,
  generateStaticMessage,
} from 'mock-builders/generator/message';
import { generateStaticUser } from 'mock-builders/generator/user';

import { MessageTextContainer } from '../MessageTextContainer';

import { defaultTheme } from '../../../../styles/themeConstants';

afterEach(cleanup);

describe('MessageTextContainer', () => {
  it('should render message text container', async () => {
    const staticUser = generateStaticUser(1);
    const message = generateMessage({
      user: { ...staticUser, image: undefined },
    });
    const { getByTestId, getByText, rerender, toJSON } = render(
      <ThemeProvider theme={defaultTheme}>
        <MessageTextContainer
          alignment='right'
          groupStyles={['top']}
          message={message}
        />
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('message-text-container')).toBeTruthy();
      expect(getByText(message.text)).toBeTruthy();
    });

    rerender(
      <ThemeProvider theme={defaultTheme}>
        <MessageTextContainer
          alignment='right'
          groupStyles={['top']}
          message={message}
          MessageText={({ message }) => (
            <Text testID='message-text'>{message.text}</Text>
          )}
        />
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('message-text-container')).toBeTruthy();
      expect(getByTestId('message-text')).toBeTruthy();
      expect(getByText(message.text)).toBeTruthy();
    });

    const staticMessage = generateStaticMessage('Hello World', {
      user: staticUser,
    });

    rerender(
      <ThemeProvider theme={defaultTheme}>
        <MessageTextContainer message={staticMessage} />
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
