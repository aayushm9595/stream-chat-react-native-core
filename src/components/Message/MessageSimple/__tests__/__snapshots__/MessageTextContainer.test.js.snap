// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageTextContainer should render message text container 1`] = `
<View
  groupStyle="undefinedBottom"
  style={
    Array [
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#EBEBEB",
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderColor": "transparent",
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "borderWidth": 0,
        "marginTop": 2,
        "paddingHorizontal": 8,
        "paddingVertical": 5,
      },
    ]
  }
  testID="message-text-container"
  type="regular"
>
  <View
    style={
      Array [
        Object {
          "minHeight": 1,
          "minWidth": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#222222",
            },
            undefined,
          ]
        }
      >
        Hello World
      </Text>
    </Text>
  </View>
</View>
`;
